### PR TITLE
Allow for instrument fields returned to be filtered by a Localization

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR allows for filtering the set of tiles returned for an instrument by a Localization. 

Co-authored by @stefanv.

Screen Shot 2022-03-11 at 6 09 20 AM

### CODE REVIEW

**Comment by mcoughlin** [comment]
@stefanv for speed, I tried also returning the contour corners as int's, just to see if that helps.

File: skyportal/tests/api/test_instrument.py [hunk 1]
```diff
@@ -46,11 +46,34 @@ def test_token_user_post_get_instrument(super_admin_token):
...
+                'GET',
+                f'instrument/{instrument_id}',
+                params=params,
+                token=super_admin_token,
+            )
+            assert status == 200
+            assert data['status'] == 'success'
+            assert data['data']['band'] == 'NIR'
+
+            print(data['data'])
```
Comment (guynir42) [inline skyportal/tests/api/test_instrument.py:68]: print statement

**Review by guynir42 — COMMENTED** [review]
Looks good. I have minor comments.

File: skyportal/handlers/api/instrument.py [hunk 2]
```diff
@@ -111,16 +115,43 @@ def get(self, instrument_id=None):
...
               description: |
-                Boolean indicating whether to include associated geojson summary bounding box. Defaults to
+                Boolean indicating whether to include associated GeoJSON summary bounding box. Defaults to
                 false.
+            - in: query
+              name: localizationDateobs
+              schema:
+                type: string
+              description: |
+                Event time in ISO 8601 format (`YYYY-MM-DDTHH:MM:SS.sss`).
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/handlers/api/instrument.py:133]: Yeah extending the docs here a bit.

File: skyportal/handlers/api/instrument.py [hunk 3]
```diff
@@ -166,38 +197,246 @@ def get(self, instrument_id=None):
...
+                ]
+            elif includeGeoJSONSummary:
+                options = [
+                    joinedload(Instrument.fields).undefer(
+                        InstrumentField.contour_summary
+                    )
+                ]
+            else:
+                options = []
+
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/handlers/api/instrument.py:229]: great catch.

File: skyportal/handlers/api/instrument.py [hunk 4]
```diff
@@ -146,15 +177,15 @@ def get(self, instrument_id=None):
...
                 false.
             - in: query
               name: includeGeoJSONSummary
               nullable: true
               schema:
                 type: boolean
               description: |
-                Boolean indicating whether to include associated geojson summary bounding box. Defaults to
+                Boolean indicating whether to include associated GeoJSON summary bounding box. Defaults to
                 false.
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/handlers/api/instrument.py:189]: good idea.

File: skyportal/tests/api/test_instrument.py (hunk 1 above)
Comment (mcoughlin) [reply skyportal/tests/api/test_instrument.py:68]: oops.

**Review by guynir42 — APPROVED** [review]
LGTM

**Comment by mcoughlin** [comment]
Thanks @guynir42!

**Comment by stefanv** [comment]
@mcoughlin Did you figure out the 90MB download issue?

**Pull request merged**